ramips: add support for TP-Link MR600 V2(EU)

linosgian opened the pull request:
MR600 V2(EU) is an LTE router that also supports 4G+ band aggregation etc. and can reportedly achieve higher bandwidth with it.

## Specifications:

* SoC: Mediatek MT7621DAT 880MHz
* RAM: 128MB DDR3
* Flash: 16MB SPI NOR flash (GD25Q128C)
* LTE Modem: Qualcomm MDM9240
* WiFi 5GHz: Mediatek MT7613BEN
* WiFi 2.4GHz: Mediatek MT7603EN
* Ethernet: MT7530, 4x 1000Base-T.
* UART: Serial console (115200 8n1), J1(GND:3)
* Buttons: Reset, WPS.
* LED: Power, WAN, LTE, WiFi 2GHz and 5GHz, LAN, Signal1, Signal2, Signal3

## MAC Addresses:

OEM firmware configuration:
54:af:97:xx:xx:7b : 2.4G
54:af:97:xx:xx:7a : 5G
54:af:97:xx:xx:7c : LTE
54:af:97:xx:xx:7b : LAN (label)
54:af:97:xx:xx:7c : WAN

## Installation:

1. Download the OpenWrt initramfs-image.

Place it into a TFTP server root directory and rename it to openwrt.img
Configure the TFTP server to listen at 192.168.0.5/24.

3. Connect to the serial console.

Attach power and interrupt the boot procedure when prompted (type `tpl`).

Credentials are admin / 1234

4. Configure U-Boot for booting OpenWrt from ram

 $ tftpboot
 $ bootm

5. Transfer the OpenWrt sysupgrade image to the device.

## LTE:

In order to setup the wwan0 interface:
1. Add a `qmi` proto interface under `/etc/config/network`, e.g.:
```
config interface 'wwan0'
        option device '/dev/cdc-wdm0'
        option proto 'qmi'
        option pincode 'XXXX'
        option apn 'your_isp_apn'
```

2. Add `wwan0` interface to the `wan` firewall zone
3. `/etc/init.d/network restart`

Signed-off-by: Linos Giannopoulos <linosgian00@gmail.com>

linosgian committed:
ramips: add support for TP-Link MR600 V2(EU)

MR600 V2(EU) is an LTE router that also supports 4G+ band aggregation
etc. and can reportedly achieve higher bandwidth with it.

- Specifications:

* SoC: Mediatek MT7621DAT 880MHz
* RAM: 128MB DDR3
* Flash: 16MB SPI NOR flash (GD25Q128C)
* LTE Modem: Qualcomm MDM9240
* WiFi 5GHz: Mediatek MT7613BEN
* WiFi 2.4GHz: Mediatek MT7603EN
* Ethernet: MT7530, 4x 1000Base-T.
* UART: Serial console (115200 8n1), J1(GND:3)
* Buttons: Reset, WPS.
* LED: Power, WAN, LTE, WiFi 2GHz and 5GHz, LAN, Signal1, Signal2,
  Signal3

- MAC Addresses:

OEM firmware configuration:
54:af:97:xx:xx:7b : 2.4G
54:af:97:xx:xx:7a : 5G
54:af:97:xx:xx:7c : LTE
54:af:97:xx:xx:7b : LAN (label)
54:af:97:xx:xx:7c : WAN

- Installation:

1. Download the OpenWrt initramfs-image.

Place it into a TFTP server root directory and rename it to openwrt.img
Configure the TFTP server to listen at 192.168.0.5/24.

3. Connect to the serial console.

Attach power and interrupt the boot procedure when prompted (type `tpl`).

Credentials are admin / 1234

4. Configure U-Boot for booting OpenWrt from ram

 $ tftpboot
 $ bootm

5. Transfer the OpenWrt sysupgrade image to the device.

- LTE:

In order to setup the wwan0 interface:
1. Add a `qmi` proto interface under `/etc/config/network`, e.g.:
```
config interface 'wwan0'
        option device '/dev/cdc-wdm0'
        option proto 'qmi'
        option pincode 'XXXX'
        option apn 'your_isp_apn'
```

2. Add `wwan0` interface to the `wan` firewall zone
3. `/etc/init.d/network restart`

Signed-off-by: Linos Giannopoulos <linosgian00@gmail.com>

obsy commented on target/linux/ramips/mt7621/base-files/etc/board.d/02_network:
```diff
@@ -96,6 +96,9 @@ ramips_setup_interfaces()
 		ucidef_set_interface_lan "lan1 lan2 lan3 plc0"
 		;;
+	tplink,mr600-v2-eu)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
```
There is no port named wan in dts?

linosgian replied:
Hey the `wan` port is covered [here](https://github.com/openwrt/openwrt/pull/10887/files#diff-6b77468626909ff021fe50c7b8fbc364245df0bc015e0fa50c7a401d7db198c7R165). I've tried to replicate what this [PR](https://github.com/openwrt/openwrt/pull/10238) did, in order to mux all ports under `gmac1` for more bandwidth.

thisboi2 commented:
can i install the image through TP-Links own local firmware upgrade page in the router settings?

adx74 commented:
Hey dude @linosgian where is the serial port?

TheRealHypR commented:
> Hey dude @linosgian where is the serial port?
look here: https://forum.openwrt.org/t/tp-link-archer-mr600-exploration/65489/15

adx74 commented:
Okay, and it is possible to flash it directly from the web panel? @linosgian

adx74 commented:
Hello, i have solder pins to the serial port, can you please send the .img to me? Thanks in advance @linosgian

adx74 commented:
@linosgian when i enter tftpboot command it asking for test.bin, can you help me please ?

TheRealHypR commented:
> @linosgian when i enter tftpboot command it asking for test.bin, can you help me please ?

Here are the two images I used with my device: https://we.tl/t-UGQEhc1Ugk
1. rename initramfs image to test.bin and use tftp to boot.
2. for example: configure a wifi network via command line then flash the 2nd image via web ui (sysupgrade)

adx74 commented:
> > @linosgian when i enter tftpboot command it asking for test.bin, can you help me please ?
> 
> Here are the two images I used with my device: https://we.tl/t-UGQEhc1Ugk
> 
> 1. rename initramfs image to test.bin and use tftp to boot.
> 2. for example: configure a wifi network via command line then flash the 2nd image via web ui (sysupgrade)

Thanks bro, can i tip you with ppl?

adx74 commented:
@TheRealHypR Hey, when i prompt `uqmi -d /dev/cdc-wdm0 --get-data-status`, it return `"disconnected"`, and on the web panel it's say at wwan0 : "Error: Unknown error (NETWORK_REGISTRATION_FAILED)".

Did you have any idea/fix? Thx in advance

linosgian commented:
Hey @adx74 sorry I missed the previous comments, and thanks @TheRealHypR for picking them up. Can we migrate the conversation about debugging the LTE part over to this thread on OpenWRT's [forum](https://forum.openwrt.org/t/tp-link-archer-mr600-exploration/65489) instead? This will help reviewers of this PR in the future. Thanks!

adx74 commented:
Yes ofc, sending a msg rn :)

hauke commented on target/linux/ramips/mt7621/base-files/etc/board.d/02_network:
```diff
@@ -96,6 +96,9 @@ ramips_setup_interfaces()
 		ucidef_set_interface_lan "lan1 lan2 lan3 plc0"
 		;;
+	tplink,mr600-v2-eu)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
```
You can configure the wwan0 port now too, see bf9dc176516bdf030486483e3b766b9f49f3db15

bam80 replied:
Maybe it's also time to decide if we need a convenient wrapper for this?
https://github.com/openwrt/openwrt/pull/10632#issuecomment-1281503138
@Leo-PL

Leo-PL replied:
@bam80 I'm in favor of this and wanted to do that, but had no time to implement it this weekend.

bam80 replied:
@Leo-PL please ping me if you do, I'll update my device to utilize the wrapper

linosgian replied:
> You can configure the wwan0 port now too, see [bf9dc17](https://github.com/openwrt/openwrt/commit/bf9dc176516bdf030486483e3b766b9f49f3db15) and #10972

I've just pushed a fixup for this, thanks for the heads up!

remipixel commented:
Hey ! Thanks for your work! Can you please re-send the img file so i can try it too ? (and pics for the teardown plz, dead link too...)

remipixel commented:
> > @linosgian when i enter tftpboot command it asking for test.bin, can you help me please ?
> 
> Here are the two images I used with my device: https://we.tl/t-UGQEhc1Ugk
> 
>     1. rename initramfs image to test.bin and use tftp to boot.
> 
>     2. for example: configure a wifi network via command line then flash the 2nd image via web ui (sysupgrade)

Hey, can you help me please ? Adx74 send me the files but im stuck at the kernel loading, after typing "bootm 83FFFE00". I was running the last TP LINK firmware version from 2022. Thx

linosgian commented:
> Hey, can you help me please ? Adx74 send me the files but im stuck at the kernel loading, after typing "bootm 83FFFE00". I was running the last TP LINK firmware version from 2022. Thx

I've just pushed a fixup. Can you try to build with it and test it out?

linosgian commented:
> > > Hey, can you help me please ? Adx74 send me the files but im stuck at the kernel loading, after typing "bootm 83FFFE00". I was running the last TP LINK firmware version from 2022. Thx
> > 
> > 
> > I've just pushed a fixup. Can you try to build with it and test it out? Mind that I've also updated the notes around booting the initramfs image (you can just `bootm` now)
> 
> Hey, i just built with the fixup, initramfs (renamed as test.bin) boots perfectly but i cant connect to any lan ports and it shows no web interface, etc. i think my build is quite different and has different packages. can you try to build yourself with your package configuration then share us please ? tnks

Can you try again now? I've rebased and noticed that the fixup was breaking the network config for you, hence no interface was coming up when you tried it out. Sorry bout that.

Imandaneshi commented:
I have this router and I really want to test this, is this safe and won't break my router?
Also my knowledge is limited to backend development, is there a guide to how to do this ?

remipixel commented:
Hey you need to solder pins to the board first than connect them through a serial console so make sure you have all theses components. I recommend you to do this to the back of the board because of heat dissipator. You have the links to the plans for RX/TX/GND connectors (you don't need 3,3V pin), higher in the PR. 

After this, everything is noted to the procedure, build your own openwrt build with Linogian's fork on the right branch and this should be right. 



> @linosgian I have this router and I really want to test this, is this safe and won't break my router?
> 
> Also my knowledge is limited to backend development, is there a guide to how to do this ?

Imandaneshi commented:
> Hey you need to solder pins to the board first than connect them through a serial console so make sure you have all theses components. I recommend you to do this to the back of the board because of heat dissipator. You have the links to the plans for RX/TX/GND connectors (you don't need 3,3V pin), higher in the PR.
> 
> After this, everything is noted to the procedure, build your own openwrt build with Linogian's fork on the right branch and this should be right.
> 
> > @linosgian I have this router and I really want to test this, is this safe and won't break my router?
> > Also my knowledge is limited to backend development, is there a guide to how to do this ?

Can I use tftp to avoid opening the device ?

hauke commented:
Thank you for the patch; I applied it to master in 78110c3b5fce119d13cd45dadd33ca396c8ce197.

remipixel commented:
> > Hey you need to solder pins to the board first than connect them through a serial console so make sure you have all theses components. I recommend you to do this to the back of the board because of heat dissipator. You have the links to the plans for RX/TX/GND connectors (you don't need 3,3V pin), higher in the PR.
> 
> > 
> 
> > After this, everything is noted to the procedure, build your own openwrt build with Linogian's fork on the right branch and this should be right.
> 
> > 
> 
> > > @linosgian I have this router and I really want to test this, is this safe and won't break my router?
> 
> > > Also my knowledge is limited to backend development, is there a guide to how to do this ?
> 
> 
> 
> Can I use tftp to avoid opening the device ?

It's not possible, please read first.

RubenKelevra commented:
Where's the Serial port? How are the pins layed out?

@TheRealHypR your linked content is offline

nitinjs commented:
How to configure the band aggregation?
once installed openwrt

s0mm3rb commented:
the serial port is hidden on the bottom, so you need to detach the wifi and lte antennas and get the pcb out
top right is TX next one down is RX and then GND
here's a pic (archer mr600 eu v2):
![archer_mr600_v2_eu_serial](https://user-images.githubusercontent.com/38328575/217382586-a508a870-77b4-4bd5-9bf9-4248d64d256b.jpg)

RafaelKa commented:
Hey!

Where are images located on OpenWrt mirrors? I can't find anything in [v. 22.03.3](https://downloads.openwrt.org/releases/22.03.3/targets/ramips/mt7621/), which is released after this patch was applied.

What are the next steps for official OpenWrt builds?

remipixel commented:
Hey, try to use snapshot channel. There is no official v22.03 support for now

De : Rafael Kähm ***@***.***>
Envoyé : samedi 11 février 2023 13:22
À : openwrt/openwrt ***@***.***>
Cc : Rémi Pixel @ Pixel Digital ***@***.***>; Manual ***@***.***>
Objet : Re: [openwrt/openwrt] ramips: add support for TP-Link MR600 V2(EU) (PR #10887)


Hey!

Where are images located on OpenWrt mirrors? I can't find anything in v. 22.03.3<https://downloads.openwrt.org/releases/22.03.3/targets/ramips/mt7621/>, which is released after this patch was applied.

What are the next steps for official OpenWrt builds?

—
Reply to this email directly, view it on GitHub<https://github.com/openwrt/openwrt/pull/10887#issuecomment-1426744044>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJMCMPNKUWSKFXV474ML4YLWW577BANCNFSM6AAAAAAQ3YFHLY>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

RafaelKa commented:
@remipixel thanks for the hint.

The https://downloads.openwrt.org/snapshots/targets/ramips/mt7621/
has the images.

firefoxOnFire commented:
> @remipixel thanks for the hint.
> 
> The https://downloads.openwrt.org/snapshots/targets/ramips/mt7621/ has the images.

hey, any tutorial to install this ?? like video or something. I am new to this thing.

V3rzeT commented:
I made a tutorial, hopefully it will help someone :>
https://youtu.be/sjkU99-2onA

s0mm3rb commented:
> I made a tutorial, hopefully it will help someone :>
> https://youtu.be/sjkU99-2onA

great video👍

all necessary steps well explained

should be helpful even for not experienced users

remipixel commented:
Thank you, very nice !

De : ***@***.***>
Envoyé le :jeudi 6 avril 2023 17:32
À : ***@***.***>
Cc : Rémi Pixel @ Pixel ***@***.***>; ***@***.***>
Objet :Re: [openwrt/openwrt] ramips: add support for TP-Link MR600 V2(EU) (PR #10887)


I made a tutorial, hopefully it will help someone :>
https://youtu.be/sjkU99-2onA

—
Reply to this email directly, view it on GitHub<https://github.com/openwrt/openwrt/pull/10887#issuecomment-1499257651>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJMCMPOO2SMTLARIMZJ763DW73OXPANCNFSM6AAAAAAQ3YFHLY>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

firefoxOnFire commented:
> I made a tutorial, hopefully it will help someone :> https://youtu.be/sjkU99-2onA

Thanks 😊👍
